pydy/viz/scene.py

```diff
@@ -408,9 +408,17 @@ def _generate_json(self, directory=None, prefix=None):
         self._generate_scene_dict()
 
         self._scene_info["simulationData"] = self._simulation_json_file
-        # NOTE : Python 3 division is imported at the top of the file so
-        # this will be a float.
-        self._scene_info["timeDelta"] = 1 / self.frames_per_second
+        if self.times is not None:
+            # Assume that times is evenly spaced and monotonic.
+            # TODO: Interpolate if times are not evenly spaced.
+            total_time = self.times[-1] - self.times[0]
+            self._scene_info["timeDelta"] = total_time / (num_time_steps - 1)
+            self._scene_info["startTime"] = self.times[0]
+        else:
+            self._scene_info["timeDelta"] = 1.0 / self.frames_per_second
```

Comment:
This is ready to be merged.
On Oct 17, 2015 16:01, "Tarun Gaba" <notifications@github.com> wrote:

> In pydy/viz/scene.py
> <https://github.com/pydy/pydy/pull/272#discussion_r42309534>:
>
> > @@ -408,9 +408,17 @@ def _generate_json(self, directory=None, prefix=None):
> >          self._generate_scene_dict()
> >
> >          self._scene_info["simulationData"] = self._simulation_json_file
> > -        # NOTE : Python 3 division is imported at the top of the file so
> > -        # this will be a float.
> > -        self._scene_info["timeDelta"] = 1 / self.frames_per_second
> > +        if self.times is not None:
> > +            # Assume that times is evenly spaced and monotonic.
> > +            # TODO: Interpolate if times are not evenly spaced.
> > +            total_time = self.times[-1] - self.times[0]
> > +            self._scene_info["timeDelta"] = total_time / (num_time_steps - 1)
> > +            self._scene_info["startTime"] = self.times[0]
> > +        else:
> > +            self._scene_info["timeDelta"] = 1.0 / self.frames_per_second
>
> @oliverlee <https://github.com/oliverlee> You should include the
> "startTime" dict key here also (can be set to zero). Else the scene_desc
> json files will become in-consistent for two cases, and js code will break.
> Then it should be okay to be merged in.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pydy/pydy/pull/272/files#r42309534>.
>